### PR TITLE
[Linter] Param's length cannot exceed 22 and name should not use `_`

### PR DESCRIPTION
**Pull request by MyronFanQiu**
- param's length cannot exceed 22
- param's name should not have `_`

image

### CODE REVIEW

**Comment by haroldrandom** [comment]
title need to refine

**Comment by haroldrandom** [comment]
BTW, is 22 a decision made from discussion?

**Comment by MyronFanQiu** [comment]
> BTW, is 22 a decision made from discussion?

Yes. We can discuss this value. https://github.com/Azure/azure-cli-dev-tools/issues/233

File: azdev/operations/linter/rules/parameter_rules.py [hunk 1]
```diff
@@ -139,3 +139,26 @@ def _help_contains_queries(help_str, queries):
...
+def option_length_too_long(linter, command_name, parameter_name):
+    min_length = None
+    length_threshold = 22  # only 5% argument's length exceed this value.
+    options_list = linter.get_parameter_options(command_name, parameter_name) or []
+    for option in options_list:
+        if isinstance(option, Deprecated) or option.startswith('--__'):
+            return
+        min_length = min(min_length, len(option)) if min_length else len(option)
+    if min_length and min_length > length_threshold:
+        raise RuleError("The lengths of all options {} are longer than {} ".format(options_list, length_threshold))
```
Comment (fengzhou-msft) [inline azdev/operations/linter/rules/parameter_rules.py:154]: So this means at least one option not exceeding the length_threshold should be provided? Can you add the rule in the error message to make it clear?

Comment (MyronFanQiu) [reply azdev/operations/linter/rules/parameter_rules.py:154]: Sure

**Pull request merged**